app/src/lib/git/for-each-ref.ts

```diff
@@ -101,3 +101,104 @@ export async function getBranches(
 
   return branches
 }
+
+/**
+ * Gets all branches that differ from their upstream (i.e. they're ahead,
+ * behind or both), excluding the current branch.
+ * Useful to narrow down a list of branches that could potentially be fast
+ * forwarded.
+ *
+ * @param repository Repository to get the branches from.
+ * @param allBranches All known branches in the repository.
+ * @param currentBranchName Name of the current branch.
+ */
+export async function getBranchesDifferingFromUpstream(
+  repository: Repository,
+  allBranches: ReadonlyArray<Branch>,
+  currentBranchName: string | null
+): Promise<ReadonlyArray<Branch>> {
+  const format = [
+    '%(refname)',
+    '%(refname:short)',
+    '%(objectname)', // SHA
+    '%(upstream)',
+    '%(symref)',
+  ].join('%00')
+
+  const prefixes = ['refs/heads', 'refs/remotes']
```

Comment:
Aside from copy&pasting 😆  it felt clearer to have this in a separated array. Happy to write these inline though.

Reply:
Love it